beacon-chain/blockchain/process_block.go

```diff
@@ -135,6 +136,9 @@ func (s *Service) onBlock(ctx context.Context, signed *ethpb.SignedBeaconBlock,
 		if err := s.stateGen.MigrateToCold(ctx, fBlock.Block.Slot, fRoot); err != nil {
 			return nil, errors.Wrap(err, "could not migrate to cold")
 		}
+
+		s.depositCache.PrunePendingDeposits(ctx, int64(postState.Eth1DepositIndex()))
```

Comment:
Can you please show me where this pruning is done? The only call to `PrunePendingDeposits` happens in `log_processing.go`, which I believe is not block processing code. There is a `ProcessDeposit` function inside `block_operations.go`, but I don't see any code removing pending deposits in there. I added this line because I couldn't find the thing that you mention.